integration/test/draining/draining_test.go

```diff
@@ -0,0 +1,249 @@
+
+		go func() {
+			for {
+				time.Sleep(200 * time.Millisecond)
+
+				select {
+				case <-done:
+					return
+				default:
+					tlsConfig, err := rest.TLSConfigFor(g.RestConfig())
+					if err != nil {
+						fmt.Printf("expected %#v got %#v", nil, err)
+						continue
+					}
+					client := &http.Client{
+						Transport: &http.Transport{
+							TLSClientConfig: tlsConfig,
+						},
+					}
+
+					restClient := g.K8sClient().Discovery().RESTClient()
+					u := restClient.Get().AbsPath("api", "v1", "namespaces", "e2e-app", "services", "e2e-app:8000", "proxy/").URL()
+					resp, err := client.Get(u.String())
+					if err != nil {
+						fmt.Printf("expected %#v got %#v\n", nil, err)
+						continue
+					} else {
+						defer resp.Body.Close()
+					}
+
+					b, err := ioutil.ReadAll(resp.Body)
+					if err != nil {
+						fmt.Printf("expected %#v got %#v\n", nil, err)
+						continue
+					}
+
+					var r E2EAppResponse
+					err = json.Unmarshal(b, &r)
+					if err != nil {
+						fmt.Printf("expected %#v got %#v\n", nil, err)
+						continue
+					}
+
+					if r.Name != "e2e-app" {
+						failure++
+					} else if r.Source != "https://github.com/giantswarm/e2e-app" {
+						failure++
+					} else {
+						success++
+					}
+				}
+			}
+		}()
+	}
+
+	{
+		newLogger.Log("level", "debug", "message", "verifying e2e-app availability 10 more seconds")
+		time.Sleep(10 * time.Second)
+
+		newLogger.Log("level", "debug", "message", "scaling down guest cluster worker")
+
+		// scale down guest cluster
+		masterCount, err := numberOfMasters(env.ClusterID())
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		workerCount, err := numberOfWorkers(env.ClusterID())
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		err = removeWorker(env.ClusterID())
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		err = g.WaitForNodesUp(masterCount + workerCount - 1)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		newLogger.Log("level", "debug", "message", "verifying e2e-app availability 10 more seconds")
+		time.Sleep(10 * time.Second)
+		close(done)
+	}
+
+	{
+		newLogger.Log("level", "debug", "message", "validating test data")
+
+		percOfFail := failure * 100 / success
+
+		newLogger.Log("level", "debug", "message", fmt.Sprintf("failure count is %f", failure))
+		newLogger.Log("level", "debug", "message", fmt.Sprintf("success count is %f", success))
+		newLogger.Log("level", "debug", "message", fmt.Sprintf("ration is %f of failures", percOfFail))
+
+		if percOfFail > 5 {
+			t.Fatalf("expected %#v got %#v", "less than 5 percent of failures", fmt.Sprintf("%f of failures", percOfFail))
+		}
+	}
+}
+
+func numberOfMasters(clusterID string) (int, error) {
+	cluster, err := h.AWSCluster(clusterID)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	return key.MasterCount(*cluster), nil
+}
```

Comment:
This stuff is copied from the scaling tests. It should actually be part of the e2e harness framework I guess.

Reply:
Yeah we can think about something. You are right that e2e-harness might become too specific when we put our provider related stuff in there.